rpc_util.go

```diff
@@ -181,6 +183,24 @@ func FailFast(failFast bool) CallOption {
 	})
 }
 
+// CallCodec returns a CallOption that uses the given Codec for the RPC call.
+// This will override the Codec assocated with the ClientConn.
+func CallCodec(codec Codec) CallOption {
+	return beforeCall(func(c *callInfo) error {
+		c.codec = codec
+		return nil
+	})
+}
+
+// CallContentSubtype returns a CallOption that will set the content subtype
+// to the given string. Normally, the value returned from Codec.String() is used.
+func CallContentSubtype(contentSubtype string) CallOption {
```

Comment:
I don't understand the concern - this is making it so the subtype does not rely on the codec. This is useful for situations where you might have a passthrough codec (as you do in your proxy code, or as we have in yarpc-go right now) where you do the encoding outside of grpc-go, but still want to set the subtype (ie what we need to do). grpc-go tightly ties the encoding to the transport, and there's no real way to decouple these at this point in terms of the grpc-go code.